ocs_ci/ocs/resources/cloud_manager.py

```diff
@@ -0,0 +1,296 @@
+        if key_id and access_key:
+            self.client = boto3.resource(
+                's3', verify=verify, endpoint_url=endpoint,
+                aws_access_key_id=key_id,
+                aws_secret_access_key=access_key
+            )
+            self.access_key = key_id
+            self.secret_key = access_key
+        else:
+            self.client = boto3.resource('s3', endpoint_url=endpoint)
+            # create a secret for the underlying storage to use
+            session = boto3.Session()
+            credentials = session.get_credentials()
+            credentials = credentials.get_frozen_credentials()
+            self.access_key = credentials.access_key
+            self.secret_key = credentials.secret_key
+        bs_secret_data = templating.load_yaml(constants.MCG_BACKINGSTORE_SECRET_YAML)
+        bs_secret_data['metadata']['name'] += f'-client-secret'
+        bs_secret_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
+        bs_secret_data['data']['AWS_ACCESS_KEY_ID'] = base64.urlsafe_b64encode(
+            self.access_key.encode('UTF-8')
+        ).decode('ascii')
+        bs_secret_data['data']['AWS_SECRET_ACCESS_KEY'] = base64.urlsafe_b64encode(
+            self.secret_key.encode('UTF-8')
+        ).decode('ascii')
+        self.secret = create_resource(**bs_secret_data)
+
+    def internal_create_uls(self, name, region=None):
+        """
+        Creates the Underlying Storage using the S3 API
+        Args:
+           name (str): The Underlying Storage name to be created
+           region (str): The region to create the Underlying Storage, if none will create at
+           `us-east-1` IMPORTANT!!! note that `us-east-1` will cause an error if used since it is
+           the default region for aws
+
+        """
+        if region is None:
+            self.client.create_bucket(Bucket=name)
+        else:
+            self.client.create_bucket(
+                Bucket=name,
+                CreateBucketConfiguration={
+                    'LocationConstraint': region
+                }
+            )
+
+    def internal_delete_uls(self, name):
+        """
+        Deletes the Underlying Storage using the S3 API
+        Args:
+           name (str): The Underlying Storage name to be deleted
+
+        """
+        # Todo: rename client to resource (or find an alternative)
+        self.client.meta.client.delete_bucket_policy(
+            Bucket=name
+        )
+        sample = TimeoutSampler(
+            timeout=30, sleep=3, func=self.wait_for_delete_uls,
+            name=name
+        )
+        if not sample.wait_for_func_status(result=True):
+            logger.error(
+                f'Deletion of Underlying Storage {name} timed out. Unable to delete {name}'
+            )
+            raise TimeoutExpiredError
+        else:
+            logger.info(f'Underlying Storage {name} deleted successfully')
+
+    def wait_for_delete_uls(self, name):
+        """
```

Comment:
add empty lines between docstring sections. Should looks like:

```python
"""
This function does this and that..

Args:
    arg1 (type):

Returns:
    type: description

"""
```

Change in all other places